crates/iceberg-catalog/src/config.rs

```diff
@@ -2,6 +2,7 @@
 use anyhow::anyhow;
 use std::collections::HashSet;
 use std::convert::Infallible;
+use std::num::NonZeroUsize;
```

Comment:
Fancy!

Reply:
it's somewhat of a pain to work with.. there's no const constructor which would be infallible that I found